File: app/src/main/java/at/shockbytes/dante/network/google/GoogleBooksApi.java
package at.shockbytes.dante.network.google;

import at.shockbytes.dante.util.books.BookSuggestion;
import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * @author Martin Macheiner
 *         Date: 13.02.2017.
 */

public interface GoogleBooksApi {

    String SERVICE_ENDPOINT = "https://www.googleapis.com/books/v1/";

    @GET("volumes")
    Observable<BookSuggestion> downloadBookSuggestion(@Query("q") String query);

}

File: app/src/main/java/at/shockbytes/dante/network/google/GoogleBooksApi.kt
package at.shockbytes.dante.network.google

import at.shockbytes.dante.books.BookSuggestion
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * @author Martin Macheiner
 * Date: 13.02.2017.
 */

interface GoogleBooksApi {

    @GET("volumes")
    fun downloadBookSuggestion(@Query("q") query: String): Observable<BookSuggestion>

    companion object {

        val SERVICE_ENDPOINT = "https://www.googleapis.com/books/v1/"
    }

}
